Dockerfile

```diff
@@ -1,26 +1,25 @@
-FROM node:8-alpine
+FROM node:12-alpine
 MAINTAINER "Dan Farrelly <daniel.j.farrelly@gmail.com>"
 
 ENV NODE_ENV production
 
 RUN apk add --no-cache curl
 
-RUN mkdir -p /usr/src/app
-WORKDIR /usr/src/app
+USER node
 
-ADD package*.json /usr/src/app/
+WORKDIR /home/node
```

Comment:
what's the benefit of hosting the application in a non-standard path?

Reply:
Because /usr/src is meant for kernel and header files. Not applications. Also having the application in the /home/node path, brings it into the node userspace. Node packages should never have root rights, to minimize chances of privilege escalation.

Reply:
home is for user, not application/services

Reply:
So what do you propose? /home/node should be fine for the application. Either way I'll have to chown the app dir to keep it usable as node user.